Sources/SotoCore/AWSEndpointDiscovery.swift

```diff
@@ -39,96 +39,52 @@ public struct AWSEndpoints {
     let endpoints: [Endpoint]
 }
 
-/// Class for storing Endpoint details
-public class AWSEndpointStorage {
-    /// endpoint url
-    internal private(set) var endpoint: String
-    /// when endpoint expires
-    internal private(set) var expiration: Date
-    /// promise for endpoint discovery process
-    private var promise: EventLoopPromise<String>?
-    /// Lock access to class
-    private let lock = NIOLock()
+/// Endpoint Storage attached to a Service
+public struct AWSEndpointStorage: Sendable {
+    let endpoint: ExpiringValue<String>
 
-    /// Initialize endpoint storage
-    /// - Parameter endpoint: Initial endpoint to use
-    public init(endpoint: String) {
-        self.expiration = Date.distantPast
-        self.endpoint = endpoint
+    public init() {
+        self.endpoint = .init(threshold: 3 * 60)
```

Comment:
Is there a reason why this is hard coded as 180?

Reply:
It is set to 3 minutes. Basically when an endpoint is about to expire it has 3 minutes to get a new one. I didn't think this was something that needed a configurable value. I've added a comment